src/_P049_MHZ19.ino

```diff
@@ -73,8 +82,37 @@ boolean Plugin_049(byte function, struct EventStruct *event, String& string)
         break;
       }
 
+    case PLUGIN_WEBFORM_LOAD:
+      {
+        byte choice = Settings.TaskDevicePluginConfig[event->TaskIndex][0];
+        String options[2] = { F("Normal"), F("ABC disabled") };
+        int optionValues[2] = { ABC_enabled, ABC_disabled };
+        addFormSelector(string, F("Auto Base Calibration"), F("plugin_049_abcdisable"), 2, options, optionValues, choice);
+        success = true;
+        break;
+      }
+
+    case PLUGIN_WEBFORM_SAVE:
+      {
+        const int formValue = getFormItemInt(F("plugin_049_abcdisable"));
+        boolean new_ABC_disable = (formValue == ABC_disabled);
+        if (Plugin_049_ABC_Disable != new_ABC_disable) {
+          // Setting changed in the webform.
+          Plugin_049_ABC_MustApply = true;
+          Plugin_049_ABC_Disable = new_ABC_disable;
+        }
+        Settings.TaskDevicePluginConfig[event->TaskIndex][0] = formValue;
+        success = true;
+        break;
+      }
+
     case PLUGIN_INIT:
       {
+        if (Plugin_049_ABC_Disable) {
+          // No guarantee the correct state is active on the sensor after reboot.
+          Plugin_049_ABC_MustApply = true;
+        }
+        Plugin_049_ABC_Disable = Settings.TaskDevicePluginConfig[event->TaskIndex][0] == ABC_disabled;
```

Comment:
i think this line should be BEFORE the if statement above?

Reply:
Yeah, looks like it. @TD-er unless you want to always disable or enable on boot? In that case you can just drop the whole MustApply logic and always set the desired state at boot based on pluginconfig or default (disable = false)

Reply:
indeed

Reply:
You are right, it is better to read the settings first.
I will make a new commit of it.